public-dashboard-client/src/controls/CohortSelect.js

```diff
@@ -0,0 +1,300 @@
+        }
+      } else {
+        newSelection = [...selected];
+
+        const index = selected.indexOf(selectedItem);
+
+        if (index === -1) {
+          newSelection.push(selectedItem);
+        } else {
+          newSelection.splice(index, 1);
+        }
+        // need to keep selection sorted or labels and colors will get out of sync
+        newSelection.sort((a, b) => ascending(a.label, b.label));
+      }
+
+      setSelected(newSelection);
+    },
+  });
+
+  useEffect(() => {
+    // index 0 is select all and should be ignored here
+    if (highlightedIndex < 1) {
+      onHighlight(undefined);
+    } else {
+      // offset by one due to select all
+      onHighlight(optionsFromData[highlightedIndex - 1]);
+    }
+  }, [highlightedIndex, onHighlight, optionsFromData]);
+
+  const labelProps = getLabelProps();
+  const toggleButtonProps = getToggleButtonProps();
+  return (
+    <>
+      <ControlLabel as="label" {...labelProps}>
+        Cohort
+      </ControlLabel>
+      <ControlValue as="button" type="button" {...toggleButtonProps}>
+        {buttonContents}
+      </ControlValue>
+      <DropdownMenu {...getMenuProps()} as="ul">
+        {isOpen &&
+          visibleOptions.map((opt, index) => {
+            const isSelected = selected.includes(opt);
+            const itemProps = getItemProps({ item: opt, index });
+            return (
+              <DropdownMenuItem
+                {...itemProps}
+                aria-selected={isSelected}
+                as="li"
+                backgroundColor={isSelected ? opt.color : undefined}
+                highlightedSelector={
+                  highlightedIndex === index ? `&#${itemProps.id}` : undefined
+                }
+                key={opt.id}
+              >
+                <MenuItemContents>
+                  {opt.label}
+                  <MenuItemCheckMark src={checkMarkPath} />
+                </MenuItemContents>
+              </DropdownMenuItem>
+            );
+          })}
+      </DropdownMenu>
+    </>
+  );
+}
+
+CustomSelect.propTypes = {
+  buttonContents: PropTypes.node.isRequired,
+  onHighlight: PropTypes.func.isRequired,
+  options: OPTIONS_PROP_TYPE.isRequired,
+  selected: OPTIONS_PROP_TYPE.isRequired,
+  setSelected: PropTypes.func.isRequired,
+};
+
+function NativeSelect({ buttonContents, options, selected, setSelected }) {
+  return (
+    <>
+      <ControlLabel aria-hidden>Cohort</ControlLabel>
+      <ControlValue aria-hidden>{buttonContents}</ControlValue>
+      <HiddenSelect
+        aria-label="Cohort"
+        multiple
+        onChange={(event) => {
+          const currentlySelectedIds = [...event.target.options]
+            .filter((opt) => opt.selected)
+            .map((opt) => opt.value);
+          setSelected(
+            options.filter((opt) => currentlySelectedIds.includes(opt.id))
+          );
+          // toggleSelected(options.find((opt) => opt.id === event.target.value));
```

Comment:
nit: Should this be removed?

Reply:
yeah whoops! thank you

Reply:
mm apparently I made this change on the other PR instead of this one but I _did_ make it ... I'm having quite a morning over here